index.rb

```diff
@@ -0,0 +1,102 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    arr = to_a
+    arr.to_enum unless block_given?
+    n = 0
+    while arr.length > n
+      yield(arr[n])
+      n += 1
+    end
+  end
+
+  def my_each_with_index
+    arr = to_a
+    return arr.to_enum unless block_given?
+
+    n = 0
+    while arr.length > n
+      yield(arr[n], n)
+      n += 1
+    end
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    res = []
+    my_each do |v|
+      res << v if yield(v) == true
+    end
+    res
+  end
+
+  def my_all?
+    check = true
+    if block_given?
+      my_each { |i| check = false unless yield(i) }
+    else
+      my_all? { |obj| obj }
+    end
+    check
+  end
+
+  def my_any?
+    check = false
+    if block_given?
+      my_each { |i| check = true if yield(i) }
+    else
+      my_any? { |obj| obj }
+    end
+    check
+  end
+
+  def my_none?
+    check = true
+    if block_given?
+      my_each { |i| check = false if yield(i) }
+    else
+      my_none { |obj| obj }
+    end
+    check
+  end
+
+  def my_count(*arg)
+    count = 0
+    unless block_given?
+      if arg.empty?
```

Comment:
- [ ] my_count returns the number of items in enum through enumeration


- [ ] my_count counts the number of items in enum that are equal to item if an argument is given

Reply:
Can you provide an example of how i should return the number of items through enumeration?